File: sandbox-runner/patterns/eager-workflow-start/go/activities.go
package main

import (
	"context"
	"errors"
	"fmt"
	"time"
)

func ValidateTransaction(_ context.Context, req TransactionRequest) (*Transaction, error) {
	if req.Amount <= 0 {
		return nil, errors.New("invalid transaction amount")
	}
	return &Transaction{
		ID:     fmt.Sprintf("tx-%d", time.Now().UnixMilli()),
		Status: "initialized",
	}, nil
}

func ReserveFunds(_ context.Context, tx *Transaction) (*Transaction, error) {
	return &Transaction{ID: tx.ID, Status: "reserved"}, nil
}

func SettleTransaction(_ context.Context, tx *Transaction) (*Transaction, error) {
	return &Transaction{ID: tx.ID, Status: "completed"}, nil
}

File: sandbox-runner/patterns/eager-workflow-start/python/activities.py
import time
from temporalio import activity
from shared import Transaction, TransactionRequest


@activity.defn
async def validate_transaction(req: TransactionRequest) -> Transaction:
    if req.amount <= 0:
        raise ValueError(f"Invalid amount: {req.amount}")
    return Transaction(id=f"tx-{int(time.time() * 1000)}", status="initialized")


@activity.defn
async def reserve_funds(tx: Transaction) -> Transaction:
    return Transaction(id=tx.id, status="reserved")


@activity.defn
async def settle_transaction(tx: Transaction) -> Transaction:
    return Transaction(id=tx.id, status="completed")
